resolver/resolver_test.go

```diff
@@ -0,0 +1,118 @@
+package resolver
+
+import (
+	"net"
+	"net/http"
+	"testing"
+)
+
+func TestResolve8888(t *testing.T) {
+	r, err := NewResolver([]string{"8.8.8.8:53"})
```

Comment:
I'd rather not call out Google every time I run the test suite :-) Can we setup a fake DNS server instead? https://godoc.org/github.com/miekg/dns

Reply:
I was trying to avoid adding deps unnecessarily.  But if you are fine with that, the thought had already crossed my mind.

Reply:
+1